mongo/gridfs/gridfs_examples_test.go

```diff
@@ -0,0 +1,225 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package gridfs_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo/gridfs"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func ExampleBucket_OpenUploadStream() {
```

Comment:
Generally, I'm finding that all the handling of the read and write deadlines (especially with the defer to change it back) is distracting from the main purpose of the examples.

Perhaps add a single extra example showing deadline usage and remove everywhere else?

Reply:
I added them to all of the examples for the same reason we duplicated comments/code in previous CRUD examples: the examples are collapsed by default and a user might only click on one of them. I doubt most people will click on multiple because the structure is very similar, so I think it's safe to leave the deadlines in all of the examples.